Sources/bip32-ed25519-swift/bip32-ed25519-swift.swift

```diff
@@ -221,21 +282,32 @@ public class Bip32Ed25519 {
         // let extPub: Data = nodePublic + nodeCC
         // let publicKey: Data = deriveChildNodePublic(extendedKey: extPub, index: bip44Path[4]).subdata(in: 0..<32)
 
-        derived = deriveChildNodePrivate(extendedKey: derived, index: bip44Path[4])
+        let g = derivationType.rawValue
+
+        var derived = rootKey
+        for i in 0 ..< bip44Path.count {
+            derived = deriveChildNodePrivate(extendedKey: derived, index: bip44Path[i], g: g)
+        }
 
-        return isPrivate ? derived : SodiumHelper.scalarMultEd25519BaseNoClamp(derived.subdata(in: 0 ..< ED25519_SCALAR_SIZE))
+        return isPrivate ? derived : SodiumHelper.scalarMultEd25519BaseNoClamp(derived.subdata(in: 0 ..< ED25519_SCALAR_SIZE)) + derived.subdata(in: 64 ..< 96)
     }
 
-    public func keyGen(context: KeyContext, account: UInt32, change: UInt32, keyIndex: UInt32) -> Data {
+    public func keyGen(
+        context: KeyContext,
+        account: UInt32,
+        change: UInt32,
+        keyIndex: UInt32,
+        derivationType: BIP32DerivationType = BIP32DerivationType.Peikert
+    ) -> Data {
         let rootKey: Data = fromSeed(seed)
         let bip44Path: [UInt32] = getBIP44PathFromContext(context: context, account: account, change: change, keyIndex: keyIndex)
 
-        return deriveKey(rootKey: rootKey, bip44Path: bip44Path, isPrivate: false)
+        return deriveKey(rootKey: rootKey, bip44Path: bip44Path, isPrivate: false, derivationType: derivationType).prefix(32)
```

Comment:
What is the point of doing .prefix(32) here? I believe it should be controlled inside deriveKey isPrivate logic